packages/flutter/lib/src/material/button.dart

```diff
@@ -589,3 +589,19 @@ class _RenderInputPadding extends RenderShiftedBox {
     );
   }
 }
+
+/// How the icon in any `button.icon` should be placed, where `button` could be:
+/// * [TextButton]
+/// * [ElevatedButton]
+/// * [OutlinedButton]
+///
+/// See also:
+///
+/// * [TextButton], [ElevatedButton], [OutlinedButton], the button widgets
+enum IconAlignment {
```

Comment:
Can I recommend naming this along the lines of something like `ButtonIconAlignment`, so as to make sense to the user in where can this enum be applied to?
Also, owing the possibility of text directionality(ltr/rtl), using `ButtonIconAlignment.left` and `ButtonIconAlignment.right` will probably be misleading(`ButtonIconAlignment.left` would display icon on the left in ltr, but right in rtl).
Something like `ButtonIconAlignment.leading/trailing` might be a better name.

Reply:
Yea, thanks a lot. It makes much sense. Will do the changes but will use `start` and `end` to make like the `MainAxisAlignment` and `CrossAxisAlignment` since people are used to those.